src/twisted/internet/endpoints.py

```diff
@@ -827,6 +836,11 @@ def __init__(
         self._hostStr = self._hostBytes if bytes is str else self._hostText
         self._port = port
         self._timeout = timeout
+        if bindAddress is not None:
+            if isinstance(bindAddress, (bytes, str)):
+                bindAddress = (bindAddress, 0)
+            if isinstance(bindAddress[0], bytes):
+                bindAddress = (nativeString(bindAddress[0]), bindAddress[1])
```

Comment:
`nativeString` is a holdover from python2 compatibility, we should not really be using it in new code.
```suggestion
                bindAddress = (bindAddress[0].decode(), bindAddress[1])
```